### ISSUE TITLE
Upgrade to Rails 7

### ISSUE THREAD

**Issue by pedantic-git** [fishpercolator/name.pn#44]
Upgrade to Rails 7 and deprecate the use of Webpack. This is a big one!

**Comment by pedantic-git** [comment]
The basic upgrade is done and passing most of the tests on the [rails7](https://github.com/fishpercolator/name.pn/tree/rails7) branch.

Still to do:

- [ ]  Fix the one failing test.
- [ ] Migrate `image_processing` from `mini_magick` to `vips` - will require changes to the Docker containers and to Heroku buildpacks.
- [ ] Remove Webpack entirely and replace it with the Rails 7 default approach. This will put us in good stead for the future!
- [ ] Update all the other dependencies.

**Comment by pedantic-git** [comment]
OK - here's where we got up to:

1. Upgraded to Rails 7
2. Have stuck with Webpacker for now but upgraded it to the more secure/maintained Shakapacker
3. Replaced most things with Rails 7 defaults, including changing the image_processing backend to vips (which requires the apt buildpack).

All this work is on the branch https://github.com/fishpercolator/name.pn/tree/rails7-attempt2

Once ActiveAdmin have a path to using jsbundling or importmaps, I'd like to revisit the following:

1. Remove webpacker and replace with either jsbundling or importmaps
2. Upgrade turbolinks to Turbo
3. Upgrade Stimulus to v3

I'll raise a case for those bits separately.

The Rails 7 env is deployed to https://name-pn-qa.herokuapp.com/ - I'll play for it for a week or so and then deploy it to production if I find no issues.
